backend/app.py

```
@@ -229,6 +229,37 @@ async def complete():
+
+@app.route('/api/document/save', methods=['POST', 'OPTIONS'])
+@validate_request
+async def save_document():
+    """Save document state"""
+    try:
+        data = await request.get_json()
+        if 'id' not in data or 'content' not in data:
+            raise AIServiceError("Missing 'id' or 'content' field", 400)
+
+        doc_id = data['id']
+        documents[doc_id] = {'content': data['content'], 'timestamp': datetime.now().isoformat()}
```

Ruff (0.8.2)

`datetime.datetime.now()` called without a `tz` argument

(DTZ005)